ray-operator/controllers/ray/rayservice_controller_unit_test.go

```diff
@@ -779,6 +782,15 @@ func TestReconcileRayCluster(t *testing.T) {
 			enableZeroDowntime:      false,
 			shouldPrepareNewCluster: true,
 		},
+		// Test 6: If active KubeRay version doesn't match the KubeRay version annotation on the RayCluster, update the RayCluster.
```

Comment:
Add more comments. For example,

// Test 6: If the active KubeRay version doesn't match the KubeRay version annotation on the RayCluster, update the RayCluster's hash and KubeRay version annotations first before checking whether to trigger a zero downtime upgrade. This behavior occurs because when we upgrade the KubeRay CRD, the hash generated by different KubeRay versions may differ, which can accidentally trigger a zero downtime upgrade.

Reply:
That test case description looks good to me, I added it in [3142ece](https://github.com/ray-project/kuberay/pull/2320/commits/3142eced7e35e4b5ca18b3f3c7aef586796b7b52).